### PR TITLE
x11-libs/startup-notification: fix incompatible pointer to integer co…

### PR DESCRIPTION
**Pull request by listout**
…nversion with clang 16

Closes: https://bugs.gentoo.org/885563

### CODE REVIEW

File: x11-libs/startup-notification/startup-notification-0.12-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit xorg-3
```
Comment (a17r) [inline x11-libs/startup-notification/startup-notification-0.12-r2.ebuild:5]: empty line after `EAPI` please.

**Review by a17r — CHANGES_REQUESTED** [review]
Thanks for your contribution. As this is a test (build) fix, there is no need for a version bump at all.

However, at the same time you make an `EAPI` bump which must be mentioned in git summary, and does require a revbbump.

Suggestion: Make it 2 commits:
- fix clang-16 (test) build error in place
- bump to EAPI-8 with revbump

File: x11-libs/startup-notification/startup-notification-0.12-r2.ebuild (hunk 1 above)
Comment (listout) [reply x11-libs/startup-notification/startup-notification-0.12-r2.ebuild:5]: Oh, I was doing this for modern c porting and hence did the revision bump

Comment (thesamesam) [reply x11-libs/startup-notification/startup-notification-0.12-r2.ebuild:5]: We need revbumps for modern C fixes usually, but this one is a test-only fix, so it doesn't change what is installed -> no revision.

File: x11-libs/startup-notification/startup-notification-0.12-r1.ebuild [hunk 2]
```diff
@@ -10,7 +11,7 @@ SRC_URI="https://www.freedesktop.org/software/${PN}/releases/${P}.tar.gz"
 
 LICENSE="LGPL-2 MIT"
 SLOT="0"
-KEYWORDS="~alpha amd64 arm arm64 ~hppa ~ia64 ~loong ~mips ppc ppc64 ~riscv sparc x86 ~amd64-linux ~x86-linux ~ppc-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
```
Comment (a17r) [inline x11-libs/startup-notification/startup-notification-0.12-r1.ebuild:14]: Don't destabilise this version please. :)

**Review by a17r — CHANGES_REQUESTED** [review]

File: x11-libs/startup-notification/startup-notification-0.12-r1.ebuild (hunk 2 above)
Comment (listout) [reply x11-libs/startup-notification/startup-notification-0.12-r1.ebuild:14]: ah, darn. I git mv-ed

Comment (listout) [reply x11-libs/startup-notification/startup-notification-0.12-r1.ebuild]: should be okay now

**Comment by mattst88** [comment]
I don't really know that this is right. It makes the thing compile, but with this warning:

```
test-send-xmessage.c:60:35: warning: cast to smaller integer type 'xcb_atom_t' (aka 'unsigned int') from 'char *' [-Wpointer-to-int-cast]                                                                                              
   60 |                                   (xcb_atom_t)argv[1],                                                                                                                                                                         
      |                                   ^~~~~~~~~~~~~~~~~~~                                                                                                                                                                          
test-send-xmessage.c:61:35: warning: cast to smaller integer type 'xcb_atom_t' (aka 'unsigned int') from 'char *' [-Wpointer-to-int-cast]                                                                                              
   61 |                                   (xcb_atom_t)argv[2],                                                                                                                                                                         
      |                                   ^~~~~~~~~~~~~~~~~~~                                                                                                                                                                          
2 warnings generated.                                                                                                                                                                                                                  
```

`test/Makefile.am` contains 

```automake
check_PROGRAMS=$(XLIB_TEST) $(XCB_TEST)
```

which causes the tests to be built with `make check` but has no `TESTS=` declaration, so they're not executed.

Even going to the hassle of wiring up the tests locally and running `make check` under `virtx`, they don't work -- some seem to require extra arguments and others hang. tl;dr: I don't think they're meant to run as part of `make check`.

**Pull request closed without merging**